extensions/markdown-language-features/src/features/workspaceSymbolProvider.ts

```diff
@@ -81,8 +90,23 @@ class VSCodeWorkspaceMarkdownDocumentProvider extends Disposable implements Work
 	}
 
 	private async getMarkdownDocument(resource: vscode.Uri): Promise<SkinnyTextDocument | undefined> {
-		const doc = await vscode.workspace.openTextDocument(resource);
-		return doc && isMarkdownFile(doc) ? doc : undefined;
+		let result = this._docIndex.getByUri(resource);
+		if (result) {
+			return await vscode.workspace.openTextDocument(resource);
+		}
+
+		let bytes = await vscode.workspace.fs.readFile(resource);
+		// We assume that markdown is in UTF-8
+		let text = new StringDecoder('utf-8').write(Buffer.from(bytes));
+		return new Promise((resolve) => {
```

Comment:
You can use `Promise.resolve(result)` directly instead of creating a `new Promise()`.